csrc/kernel_ir.h

```diff
@@ -378,6 +379,37 @@ class MBarrierArrive final : public Expr {
   }
 };
 
+class MBarrierArriveExpectTx final : public Expr {
```

Comment:
This is something new, can we add a comments stating when this will be used and how it works?

Reply:
added